### ISSUE TITLE
Get-AzRoleAssignment : The Principal ID xxxxx_live.com#EXT#@xxxxxx.onmicrosoft.com is not valid. Principal ID must be a GUID

### ISSUE THREAD

**Issue by halfluke** [hausec/PowerZure#7]
Hmm, getting a different error though:
Get-AzRoleAssignment : The Principal ID xxxxx_live.com#EXT#@xxxxxx.onmicrosoft.com is not valid. Principal ID must be a GUID

_Originally posted by @halfluke in https://github.com/hausec/PowerZure/issues/6#issuecomment-696450095_

**Comment by cyberfreaq** [comment]
I have the same issue right now.. Any updates on this?

**Comment by cyberfreaq** [comment]
The above problem can be solved when changing `$userid=$user.userprincipalname` on line 157 to `$userid=$user.id`.

However, when working with guest users i stumbled over another problem when running `Get-AzureTargets`: 
`Get-AzADUser : Cannot validate argument on parameter 'UserPrincipalName'. The argument is null or empty.`

I think the problem when working with guest users is that the user Id from Get-AzContext (e.g. cyberfreaq@auditor.onmicrosoft.com) does not correspond with the actual userPrincipalName in the AzureAD (e.g. cyberfreaq_auditor.onmicrosoft.com#EXT@customer.onmicrosoft.com).

Therefore when the following code is running (line 1512 onwards), `$user` is empty and therefore `$userid` is also empty:
```
$Context = Get-AzContext
$user = Get-AzADUser -UserPrincipalName $Context.Account
$userid=$user.id
```

Empty output:
```
PS C:\> $user
PS C:\> $userid
PS C:\> $Context

Name                                     Account                   SubscriptionName          Environment              TenantId
----                                     -------                   ----------------          -----------              --------
XXXXXXX                                  XXXXXX                    XXXXXXXXXX                XXXXXXXXXX               XXXXXXXXX
```

A workaroung could be to use the same method as used in line 138 onwards, but I didn't have time to test this yet:
```
$APSUser = Get-AzContext
$Headers = Get-AzureGraphToken 
$username = $APSUser.Account
$user = Invoke-RestMethod -Headers $Headers -Uri 'https://graph.microsoft.com/beta/me'
$userid=$user.id # The one I changed from userprincipalname to id
```

**Comment by halfluke** [comment]
Yeah, it looks like we are swinging between https://github.com/hausec/PowerZure/issues/6 and this https://github.com/hausec/PowerZure/pull/9

**Comment by cyberfreaq** [comment]
I've changed the assignment of `$user` and `$userid` to the `Invoke-RestMethod` approach in the following functions:
```
function Show-AzureCurrentUser
function Get-AzureTargets
```

Have not had any errors so far.

**Comment by halfluke** [comment]
> I've changed the assignment of `$user` and `$userid` to the `Invoke-RestMethod` approach in the following functions:
> 
> ```
> function Show-AzureCurrentUser
> function Get-AzureTargets
> ```
> 
> Have not had any errors so far.

Any chance you can open a Pull request if your fix works?

**Comment by cyberfreaq** [comment]
I've actually tried to do this directly via the browser but Github tells me "We’ve detected the file has mixed line endings. When you commit changes we will normalize them to Windows-style (CRLF)." So it seems like I made hundreds of changes although I just changed 3 lines - could imagine that this is would be more a burden for the maintainer rather than helpful.

But you asked for it so here we go :)
- Pull request: https://github.com/hausec/PowerZure/pull/10
- https://github.com/cyberfreaq/PowerZure/tree/patch-1

**Comment by halfluke** [comment]
Thanks a lot.

**Comment by hausec** [comment]
Fixed in #10
